Bomb.cpp

```diff
@@ -5,15 +5,14 @@ void Bomb::pickBlockType(Point offset) {
 	center.setPosition(offset.getX(), offset.getY());
 }
 
-
-void Bomb::draw() {
+void Bomb::draw() const {
 	drawShape('@');
 }
 
 void Bomb::explode(vector<vector<int>>* field, int offsetX, int offsetY, int height, int width)
 {
-	int x = this->center.getX() - 4;
-	int y = this->center.getY() - 4;
+	int x = max(this->center.getX() - 4,0);
```

Comment:
in case the bomb explodes at high point we get negative values on array[ ]

the minimal index to start check is 0 on height and width.
also naming could be more detailed what is x and y, new_x offsetx etc...